src/app/vm/vm-creation/template/agreement/vm-creation-agreement.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from '@angular/material';
+import { Http } from '@angular/http';
+
+import { Converter } from 'showdown';
+import { TemplateTagService } from '../../../../shared/services/tags/template-tag.service';
+import { BaseTemplateModel } from '../../../../template/shared/base-template.model';
+import { UserTagService } from '../../../../shared/services/tags/user-tag.service';
+
+
+@Component({
+  selector: 'cs-vm-creation-template-agreement',
+  templateUrl: 'vm-creation-agreement.component.html'
+})
+export class VmCreationAgreementComponent implements OnInit {
+  private _agreement: string;
+  private template: BaseTemplateModel;
+  private lang: string;
+
+  public get agreement(): string {
+    return this._agreement;
+  }
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) data,
+    private dialog: MatDialog,
+    private dialogRef: MatDialogRef<VmCreationAgreementComponent>,
+    private http: Http,
+    private templateTagService: TemplateTagService,
+    private userTagService: UserTagService
+  ) {
+    this.template = data;
+  }
+
+  public ngOnInit() {
+    this.userTagService.getLang().subscribe(res => this.lang = res);
```

Comment:
no need to store language. Use switchMap here https://github.com/bwsw/cloudstack-ui/pull/659/files#diff-bc238bfc0a14bcaff555df114d5b1d2fR49